src/SmartComponents/Recs/helpers.js

```diff
@@ -72,3 +72,34 @@ export const bulkHostActions = async ({
     });
   }
 };
+
+export const edgeSystemsCheck = async (
+  ruleId,
+  setSystemsCount,
+  setEdgeSystemsCount,
+  setConventionalSystemsCount
+) => {
+  let count = 0;
+  try {
+    await axios
+      .get(
+        `/api/insights/v1/rule/${ruleId}/systems_detail/?filter[system_profile][host_type][nil]=true`
+      )
+      .then(({ data }) => {
+        count = count += data.meta.count;
+        setConventionalSystemsCount(data.meta.count);
+      });
+    await axios
+      .get(
+        `/api/insights/v1/rule/${ruleId}/systems_detail/?filter[system_profile][host_type][edge]=true`
```

Comment:
@adonispuente I missed this: the filter syntax should be 
```suggestion
        `/api/insights/v1/rule/${ruleId}/systems_detail/?filter[system_profile][host_type]=edge`
```

Currently, the API request always returns 0 immutable count. Also, it looks like there should be a loading state as the total number stays in 0 for a while and then gets updated after API requests are finished